compiler/rustc_span/src/source_map.rs

```diff
@@ -267,26 +260,27 @@ impl SourceMap {
         self.files.borrow().stable_id_to_source_file.get(&stable_id).cloned()
     }
 
-    fn allocate_address_space(&self, size: usize) -> Result<usize, OffsetOverflowError> {
-        let size = u32::try_from(size).map_err(|_| OffsetOverflowError)?;
-
-        loop {
-            let current = self.used_address_space.load(Ordering::Relaxed);
-            let next = current
-                .checked_add(size)
-                // Add one so there is some space between files. This lets us distinguish
-                // positions in the `SourceMap`, even in the presence of zero-length files.
-                .and_then(|next| next.checked_add(1))
-                .ok_or(OffsetOverflowError)?;
-
-            if self
-                .used_address_space
-                .compare_exchange(current, next, Ordering::Relaxed, Ordering::Relaxed)
-                .is_ok()
-            {
-                return Ok(usize::try_from(current).unwrap());
-            }
-        }
+    fn register_source_file(
+        &self,
+        mut file: SourceFile,
+    ) -> Result<Lrc<SourceFile>, OffsetOverflowError> {
+        let file_id = StableSourceFileId::new(&file);
+
+        let mut files = self.files.borrow_mut();
+
```

Comment:
We need to check if `file_id` is in `stable_id_to_source_file` already inside the lock.

Reply:
Why did you mark this as resolved?

Reply:
I misread the comment. I thought it needed to be "checked in the lock", not "checked at all"

Reply:
It sounds like you read it correctly. It needs to be checked in the lock (a pre-existing issue). That doesn't need to block the PR from landing though.